src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal && animal.hasOwnProperty('hidden') && animal.hidden === false) {
```

Comment:
Better check is this animal instance of Herbivore instead of `animal.hasOwnProperty('hidden')`
```suggestion
    if (animal && animal.hasOwnProperty('hidden') && !animal.hidden) {
```